### PR TITLE
Security headers

### PR DESCRIPTION
**Pull request by RuslanHamidullin**
This is fix for "[Report Only] Refused to load the font ..." error message on Chrome/FF console
By default, we do not specify a font-src which means the value should have taken the value of default-src ('none' in this case)

Below we add "font-src data:;" to whitelist the font being loaded.
```
  config.csp = {
    default_src: "https: inline eval",
    img_src: "https:",
    font_src: "'self' data:" # font_src - defines valid sources of fonts. (http://content-security-policy.com/)
  }
```

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -6,9 +6,13 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
   # TODO: remove back once go live
-  # unless (Rails.env.test? || Rails.env.development?)
-  #   ensure_security_headers
-  # end
+  unless (Rails.env.test? || Rails.env.development?)
```
Comment (matthewford) [inline app/controllers/application_controller.rb]: Don't use parentheses around the condition of an `unless`.

**Comment by RuslanHamidullin** [comment]
@matthewford can you please take a look ^

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (matthewford) [inline app/controllers/application_controller.rb]: could we make this just `if Rails.env.production`?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -6,9 +6,13 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
   # TODO: remove back once go live
```
Comment (matthewford) [inline app/controllers/application_controller.rb]: remove comments

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -6,9 +6,13 @@ class ApplicationController < ActionController::Base
...
-  # unless (Rails.env.test? || Rails.env.development?)
-  #   ensure_security_headers
-  # end
+  unless (Rails.env.test? || Rails.env.development?)
+    # Checking on ENV["AWS_ACCESS_KEY_ID"] exists
+    # In order to avoid exceptions on dev and demo servers too
+    # as they also use production env
+
+    ensure_security_headers if ENV["AWS_ACCESS_KEY_ID"]
```
Comment (matthewford) [inline app/controllers/application_controller.rb]: I would prefer a different key, perhaps we need a new env variable?

**Comment by RuslanHamidullin** [comment]
@matthewford agree, gonna correct

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -5,10 +5,7 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
-  # TODO: remove back once go live
-  # unless (Rails.env.test? || Rails.env.development?)
-  #   ensure_security_headers
-  # end
+  ensure_security_headers if ENV['ENSURE_SECURITY_HEADERS']
```
Comment (matthewford) [inline app/controllers/application_controller.rb]: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

**Comment by RuslanHamidullin** [comment]
Need to fix fileupload issue at first:
fileupload_issue

**Comment by matthewford** [comment]
Yeah seems its the iframe upload issue - but its the same domain so in theory should be fine...

**Comment by RuslanHamidullin** [comment]
yeah,
I replaced:
```
config.csp = {
    default_src: "https: inline eval",
```
with:
```
config.csp = {
    default_src: "https: self"
```
Probably should fix. Gonna test

**Comment by RuslanHamidullin** [comment]
Nope 
```
config.csp = {
    default_src: "https: self"
```
Causes:
nope

Gonna debug

**Comment by RuslanHamidullin** [comment]
Alright, fixed iframe issue - now files are uploading.
But still some warnings. DEbugging
upload_issues

**Pull request closed without merging**